pkg/mover/intermediate.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package mover
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/internal"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+// saveIntermediateBundle will tar in this order:
+// - The original chart
+// - The hits file
+// - The container images detected as references in the chart
+//
+// The hints file goes first in the tar, followed by the chart files.
+// Finally, images are appended using the go-containerregistry tarball lib
+func saveIntermediateBundle(cd *ChartData, tarFile string, log Logger) error {
+	tmpTarfile := fmt.Sprintf("%s.tmp", tarFile)
```

Comment:
Why don't we use an actual tempfile so if smth goes wrong it stays in the temp directory and cleaned up by the OS? https://pkg.go.dev/io/ioutil#TempFile

Reply:
You are right.... yet this complicates things a bit. Because the temp file might be on another fs and I do not think a simple rename would work.

So the process is:

1. Create temp file.
2. Copy temp file to final destination with a temp name.
3. Rename temp name to final name.

2&3 are for atomicity.

will fix

Reply:
sent a fix